core_backend/app/users/models.py

```diff
@@ -74,6 +74,28 @@ async def save_user_to_db(
     return content_db
 
 
+async def update_user_retrieval_key(
+    user_db: UserDB,
+    new_retrieval_key: str,
+    asession: AsyncSession,
+) -> UserDB:
+    """
+    Updates a user's API key
+    """
+
+    # test if new_retrieval_key is a 32-character string
+    if len(new_retrieval_key) != 32:
```

Comment:
Any reason it would not be? We generate it right and not the user?

Reply:
You're right, this is probably redundant. Will remove